src/opentimelineio/imageSequenceReference.cpp

```diff
@@ -0,0 +1,174 @@
+#include "opentimelineio/imageSequenceReference.h"
+
+namespace opentimelineio { namespace OPENTIMELINEIO_VERSION  {
+
+ImageSequenceReference::ImageSequenceReference(std::string const& target_url_base,
+                      std::string const& name_prefix,
+                      std::string const& name_suffix,
+                      int start_frame,
+                      int frame_step,
+                      double const rate,
+                      int frame_zero_padding,
+                      MissingFramePolicy const missing_frame_policy,
+                      optional<TimeRange> const& available_range,
+                      AnyDictionary const& metadata)
+    : Parent(std::string(), available_range, metadata),
```

Comment:
Why can't an instance of `ImageSequenceReference` be initialized with a `name`?

Reply:
Great point. I'll see if anyone has a case for not adding that to the constructor.

Reply:
Ahh, interesting. I think I matched the constructor for `ExternalReference`. I'm guessing the main reason the api is structured that way is that `name` isn't required to be specified to have the Reference behave properly, but many of these other args are. Perhaps in python `name` should be a defaulted keyword argument toward the end of the args. Perhaps we should consider it a keyword-only argument in python 3+, but that could add to the breakage pain as studios move from python 2 to 3.

Reply:
In practice, Clips usually have names, but MediaReferences often do not. When MediaReferences do have names, it is often the same as the parent Clip, or the basename of the target_url. Therefore, it makes sense to keep it optional, and less prominent.

Reply:
I think maybe if we decide to add the name arg, we should do it in a separate PR and add it to all `MediaReference` subclasses in a similar way. `name` is still in the schema, so it's an API issue, not a schema issue.

@ssteinbach @jminor What do you think?

Reply:
My previous reply was made on an un-refreshed browser tab and I missed your last message.
I'm going to resolve this and stay consistent with `ExternalReference`. If we feel we want to address it we can examine the two together and make a decision.